tests/test_data_transform.py

```diff
@@ -78,3 +79,62 @@ def test_print_transform_gen(self):
 
         t = T.RandomFlip()
         self.assertTrue(str(t) == "RandomFlip()")
+
+    def test_random_apply_prob_out_of_range_check(self):
+        # GIVEN
+        test_probabilities = {0.0: True, 0.5: True, 1.0: True, -0.01: False, 1.01: False}
+
+        # WHEN
+        for given_probability, is_valid in test_probabilities.items():
+            # THEN
+            if not is_valid:
+                self.assertRaises(AssertionError, T.RandomApply, None, prob=given_probability)
+            else:
+                T.RandomApply(None, prob=given_probability)
+
+    def test_random_apply_wrapping_transform_gen_probability_occured_evaluation(self):
+        # GIVEN
+        test_probability = 0.001
+        transform_mock = mock.MagicMock(name="MockTransform", spec=T.TransformGen)
+        image_mock = mock.MagicMock(name="MockImage")
+        random_apply = T.RandomApply(transform_mock, prob=test_probability)
+
+        # WHEN
+        with mock.patch.object(random_apply, "_rand_range") as rand_range_mock:
+            rand_range_mock.return_value = 0.0001
```

Comment:
to simplify:
```suggestion
        with mock.patch.object(random_apply, "_rand_range", return_value=0.0001):
```
same for other tests